timelock-impl/src/main/java/com/palantir/atlasdb/backup/AtlasBackupResource.java

```diff
@@ -117,6 +121,45 @@ private InProgressBackupToken prepareBackup(Namespace namespace) {
                 .build();
     }
 
+    @Override
+    public ListenableFuture<RefreshBackupResponse> refreshBackup(AuthHeader authHeader, RefreshBackupRequest request) {
+        return handleExceptions(() -> refreshBackupInternal(authHeader, request));
+    }
+
+    private ListenableFuture<RefreshBackupResponse> refreshBackupInternal(
+            AuthHeader authHeader, RefreshBackupRequest request) {
+        if (!authHeaderValidator.suppliedTokenIsValid(authHeader)) {
+            log.error(
+                    "Attempted to complete backup with an invalid auth header. "
+                            + "The provided token must match the configured permitted-backup-token.",
+                    SafeArg.of("request", request));
+            throw new ServiceException(ErrorType.PERMISSION_DENIED);
+        }
+
+        Map<InProgressBackupToken, ListenableFuture<Optional<RefreshLockResponseV2>>> futureMap =
+                request.getTokens().stream().collect(Collectors.toMap(token -> token, this::refreshBackupAsync));
+        ListenableFuture<Map<InProgressBackupToken, RefreshLockResponseV2>> singleFuture =
+                AtlasFutures.allAsMap(futureMap, MoreExecutors.newDirectExecutorService());
+
+        return Futures.transform(singleFuture, this::getRefreshedTokens, MoreExecutors.directExecutor());
+    }
+
+    private ListenableFuture<Optional<RefreshLockResponseV2>> refreshBackupAsync(InProgressBackupToken token) {
+        Namespace namespace = token.getNamespace();
+        return Futures.transform(
+                timelock(namespace).refreshLockLeases(ImmutableSet.of(token.getLockToken())),
+                Optional::of,
+                MoreExecutors.directExecutor());
+    }
+
+    private RefreshBackupResponse getRefreshedTokens(Map<InProgressBackupToken, RefreshLockResponseV2> responses) {
+        Set<InProgressBackupToken> refreshedTokens = KeyedStream.stream(responses)
+                .filter(response -> !response.refreshedTokens().isEmpty())
+                .keys()
+                .collect(Collectors.toSet());
+        return RefreshBackupResponse.of(refreshedTokens);
+    }
```

Comment:
the responses are still `InProgressBackupTokens`, so I'm inclined to keep the wording of the response object, but rename the variable to `refreshedBackupTokens`.

I generally prefer to standardise the response object naming based on the endpoint rather than the content, as it makes it easier to extend the response object in future (e.g. adding metadata).